lib/lte_link_control/lte_lc.c

```diff
@@ -814,13 +814,22 @@ int lte_lc_normal(void)
 
 int lte_lc_psm_param_set(const char *rptau, const char *rat)
 {
-	if (rptau == NULL || strlen(rptau) != 8 ||
-		rat == NULL || strlen(rat) != 8) {
-		return -EINVAL;
+	if (rptau != NULL && strlen(rptau) == 8) {
+		memcpy(psm_param_rptau, rptau, sizeof(psm_param_rptau));
+		LOG_DBG("RPTAU set to %s", log_strdup(psm_param_rptau));
+		if (rat != NULL && strlen(rat) == 8) {
+			memcpy(psm_param_rat, rat, sizeof(psm_param_rat));
+			LOG_DBG("RAT set to %s", log_strdup(psm_param_rat));
+		} else {
+			memset(psm_param_rat, 0x00, sizeof(psm_param_rat));
+			LOG_DBG("RAT use default");
+		}
+	} else {
+		memset(psm_param_rptau, 0x00, sizeof(psm_param_rptau));
+		memset(psm_param_rat, 0x00, sizeof(psm_param_rat));
+		LOG_DBG("RPTAU & RAT use default");
```

Comment:
The documentation needs to be updated to reflect the changes in functionality. The same applies for `lte_lc_psm_req()`, and `lte_lc_psm_req()`. Also, the functions don't anymore return negative error codes, so that should be changed in the docs, or possibly convert to void function as the return code is always `0` anyway

Reply:
Updated the header file, still return -EINVAL if parameter check fails